Source/DafnyRuntime/DafnyRuntimeRust/src/lib.rs

```diff
@@ -784,29 +822,59 @@ where
                 left,
                 right,
             } => {
+                #[cfg(feature = "sync")]
```

Comment:
`cfg-if` even more appealing here IMO

Reply:
I'm grouping statements together and do some renaming so that using cfg-if will look a bit less appealing :-)